pkg/lib/saml/service.go

```diff
@@ -123,7 +123,11 @@ func (s *Service) ParseAuthnRequest(serviceProviderId string, input []byte) (*Au
 
 	// TODO(saml): Verify the signature
 
-	// TODO(saml): Verify the destination
+	if authnRequest.Destination != "" {
+		if authnRequest.Destination != s.Endpoints.SAMLLoginURL(sp.ID).String() {
+			return nil, fmt.Errorf("unexpected destination")
```

Comment:
I think we should return named errors / SAML Errors. I guess the error needs to be reported to ACS URL?